datasets/plugins/batch/batch_dataset_plugin.py

```diff
@@ -58,11 +69,39 @@ def _get_path_filters_columns(
             filters = [("run_id", "=", query_run_id)]
         return path, filters, read_columns
 
+    def read(
+        self, columns: Optional[str] = None, run_id: Optional[str] = None, **kwargs
+    ) -> Union[pd.DataFrame, "ps.DataFrame", "dd.DataFrame"]:
```

Comment:
Q1 because of implications we don't want to rush.

Reply:
Default read/write for plugin:
* `read_X()`: pandas, SparkDF, SparkPandasDF.
* not `dataset.read()` (not rush this) instead have `dataset.read_X()` (

Paved Path Plugins:
* `dataset.write(df)` can be supported without conversion, ie calls `dataset.write_X(df)` based on type